### PR TITLE
Terraform Development - Apigee Module

### PR DESCRIPTION
**Pull request by mahakverma-tudip**
- Added terraform files in dev and example folders in apigee
- Tested this module locally and it is working as expected

### CODE REVIEW

**Comment by mahakverma-tudip** [comment]
Hi @rosera 
I have created an apigee module with the required configurations. 
Some labs' DM (for example [CBL369](https://github.com/CloudVLab/gcp-training-content/blob/main/labs/CBL369-ApigeeX-UsingTargetServers/dm/lab_resources.json)) also consists of json file which is not added in this PR. Can we add parameter in the metadata for that as well? 
Thanks!

File: solutions/apigee/dev/main.tf [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    subnetwork = var.gce_machine_network == "default" ? null : var.gce_machine_network
+
+    access_config {
+      // Ephemeral IP
+    }
+  }
+
+  # Add Key/Value pair e.g. SSH keys here
+  metadata = {
+    childProjectZone            = "us-east1-b"
```
Comment (rosera) [inline solutions/apigee/dev/main.tf:27]: Is hardcoding of zone required?

**Comment by mahakverma-tudip** [comment]
Hi @rosera 
I have updated main.tf and updated hard coded zone and region to variable values. 
Please review the changes. Thanks!

**Review by rosera — CHANGES_REQUESTED** [review]
1. Please remove the hidden files from the submission.

- [ ] solutions/apigee/.DS_Store
- [ ] dev/.DS_Store
- [ ] example/.DS_Store

2. The serviceaccount.json is empty - what is the purpose of this file?

**Comment by mahakverma-tudip** [comment]
Hi @rosera 
I have Removed .DS_Store files and serviceaccount.json
Please review the changes. Thanks!

**Comment by mahakverma-tudip** [comment]
Hi @rosera 
Please merge the PR as we have updated all the required changes. Thanks!

File: solutions/apigee/dev/main.tf (hunk 1 above)
Comment (rosera) [reply solutions/apigee/dev/main.tf]: @mahakverma-tudip Please update the metadata object to use variables. This object needs to be set per use case, so should not include hardcoded properties.

**Review by rosera — CHANGES_REQUESTED** [review]
* Please update the metadata object to use variables.
* Please remove unnecessary hidden files:

- [x] solutions/apigee/.DS_Store
- [x] dev/.DS_Store
- [x] example/.DS_Store

File: solutions/apigee/example/main.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module "gce_instance" {
+  source = "github.com/CloudVLab/terraform-lab-foundation//solutions/apigee/dev"
+
+  # Pass values to the module
+  gcp_project_id = var.gcp_project_id
+  gcp_region     = var.gcp_region
+  gcp_zone       = var.gcp_zone
+  # depends_on     = []
```
Comment (rosera) [inline solutions/apigee/example/main.tf:8]: Should the metadata values not be passed in by the user of the module?

File: solutions/apigee/example/variables.tf [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+variable "apigee_name" {
+  type        = string
+  description = "Apigee Network Name."
+  default     = "apigee-network"
+}
+
+variable "region_name" {
+  type        = string
+  description = "Apigee Region Name."
+  default     = "us-central1"
```
Comment (vishal84) [inline solutions/apigee/example/variables.tf:39]: I'm assuming the same applies to the region being hard coded as stated above @rosera ?

Comment (rosera) [reply solutions/apigee/example/variables.tf:39]: @vishal84 Yep this module will be used across 30+ labs, so it should be using variables for module properties.

File: solutions/apigee/example/main.tf (hunk 2 above)
Comment (mahakverma-tudip) [reply solutions/apigee/example/main.tf:8]: Hi @rosera 
Please let us know what action to take here? Thanks!

**Comment by mahakverma-tudip** [comment]
Hi @rosera 
I have updated hard coded values to variables and have removed default values given for region and zone. Please review the changes.

File: solutions/apigee/example/main.tf (hunk 2 above)
Comment (rosera) [reply solutions/apigee/example/main.tf:8]: @mahakverma-tudip I believe the intention is to pass metadata values, therefore these should be defined in this variable?

**Pull request closed without merging**